cloudify_awssdk/common/connection.py

```diff
@@ -37,9 +38,16 @@ def __init__(self, node, aws_config=None):
             'aws_access_key_id', 'aws_secret_access_key', 'region_name']
         self.aws_config = node.properties.get(AWS_CONFIG_PROPERTY, dict())
         # Merge user-provided AWS config with generated config
-        self.aws_config.update(aws_config or dict())
+        if aws_config and type(aws_config) is dict:
```

Comment:
I disagree with this.
If `aws_config` is *not* a `dict`, then it's something that I really really want to know about because it's a violation of the contract between the plugin and the user.
With the current code, processing just continues.

We currently can't validate that an input is a `dict` at the DSL layer. This should be done in the code.

Reply:
I did update the code and validate the type of the `aws_config` outside of this module and make sure to check if it is not a valid dict

Reply:
I can't see any handling of the case when `aws_config` is defined but is not a `dict`. We need to raise an error in such case.